src/track/trackrecord.cpp

```diff
@@ -47,4 +47,85 @@ bool TrackRecord::updateGlobalKeyText(
     return false;
 }
 
+namespace {
+
+// Data migration: Reload track total from file tags if not initialized
+// yet. The added column "tracktotal" has been initialized with the
+// default value "//".
+// See also: Schema revision 26 in schema.xml
+const QString kReloadTrackTotal = "//";
+
+void mergeReplayGainMetadataProperty(
+        ReplayGain& mergedReplayGain,
+        const ReplayGain& importedReplayGain) {
+    // Preserve the values calculated by Mixxx and only merge missing
+    // values from the imported replay gain.
+    if (!mergedReplayGain.hasRatio()) {
+        mergedReplayGain.setRatio(importedReplayGain.getRatio());
+    }
+    if (!mergedReplayGain.hasPeak()) {
+        mergedReplayGain.setPeak(importedReplayGain.getPeak());
+    }
+}
+
+template<typename T>
+void mergeNullableMetadataProperty(
```

Comment:
I think this is a plain conditional copy, not a merge. Merge implies that a part is kept. 
copyIfNotNull() or so?